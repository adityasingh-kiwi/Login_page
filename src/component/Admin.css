.button-add-question {
    background-color: #4CAF50;
    border: none;
    color: white;
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    cursor: pointer;
  }

  .button-view-score {
    background-color: #008CBA;
    border: none;
    color: white;
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    cursor: pointer;
  }
  .background-admin{
      position: relative;
      /* top: 250px;
      left:40%; */
      padding: 200px 451px;
      
  }
  .background-image{
    background-image: url("/home/kiwitech/Desktop/Project/login_page/src/image/Survey.jpg");
    height: 611px;
    background-size: cover;
  }

   .addquestion-question{
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    box-sizing: border-box;
  }
  .addquestion-choices{
    width: 100%;
    padding: 15px 20px;
    margin: 8px 0;
    box-sizing: border-box;
  }
  .addquestion-answer{
    width: 100%;
    padding: 15px 20px;
    margin: 8px 0;
    box-sizing: border-box;
  }