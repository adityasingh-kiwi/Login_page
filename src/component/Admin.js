import React ,{useState} from 'react';
import './Admin.css';
import { Addquestion } from "./Addquestion";
import { useNavigate } from "react-router-dom";
import { genericFalse,genericTrue } from './Const';



 const Admin = () => {

    const [isopen ,setisopen] = useState(genericFalse);


    const handleOpenModal =()=>{
        setisopen(genericTrue) 
    }

    const isOpenClose =(data) =>{
        setisopen(data)
    }
    let Naviagte = useNavigate();
    const handleView=()=>{
        Naviagte("/viewscore");


    }


    return (
        <div className="background-image"> 
        <div className="background-admin">
            <button className="button-add-question" onClick={handleOpenModal}> Add Question</button>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <button className="button-view-score" onClick={handleView}>View Score</button>
        </div>

        <Addquestion   isopen ={isopen} isOpenClose ={(data)=>isOpenClose(data)}/>
        </div>
    )
}
export default Admin
