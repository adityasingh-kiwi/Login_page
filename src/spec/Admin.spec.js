import Admin from '../component/Admin';



describe('Admin_page',()=>{
   it('addQuestion_button check',()=>{
        let wrapper= shallow(<Admin/>);
        
        expect(wrapper.exists('.button-add-question')).toEqual(true);
        expect(wrapper.length).toEqual(1)
        wrapper.find('.button-add-question').simulate('click');
   
   })
   it('viewScore_button check',()=>{
    let wrapper= shallow(<Admin/>);
    
    expect(wrapper.exists('.button-view-score')).toEqual(true);
    expect(wrapper.length).toEqual(1)
    wrapper.find('.button-view-score').simulate('click');

})
it('Snapshot_Admin',()=>{
    let wrapper=shallow(<Admin/>)
    expect(wrapper).toMatchSnapshot();


  })
  it("Admin_button_test",()=>{
    let wrapper=shallow(<Admin/>)
    expect(wrapper.exists('#view')).toEqual(true);
     wrapper.find('#view').simulate('click');
     expect(
         wrapper.find('#view').prop('defaultValue'),
         ).toBeTruthy();
     })
     it("Admin_button_test",()=>{
      let wrapper=shallow(<Admin/>)
      expect(wrapper.exists('#add')).toEqual(true);
       wrapper.find('#add').simulate('click');
       expect(
           wrapper.find('#add').prop('defaultValue'),
           ).toBeUndefined();
       })

})