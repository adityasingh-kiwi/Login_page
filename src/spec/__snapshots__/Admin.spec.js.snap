// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin_page Snapshot_Admin 1`] = `
<div
  className="background-image"
>
  <div
    className="background-admin"
  >
    <button
      className="button-add-question"
      defaultvalue={false}
      id="add"
      onClick={[Function]}
    >
       Add Question
    </button>
           
    <button
      className="button-view-score"
      defaultValue={false}
      id="view"
      onClick={[Function]}
    >
      View Score
    </button>
  </div>
  <Addquestion
    isOpenClose={[Function]}
    isopen={false}
  />
</div>
`;
